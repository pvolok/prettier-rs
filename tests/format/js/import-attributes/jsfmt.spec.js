run_spec(import.meta, ["babel", "typescript"], {
  errors: {
    acorn: [
      "dynamic-import.js",
      "empty.js",
      "multi-types.js",
      "static-import.js",
      "re-export.js",
      "without-from.js",
      "non-type.js",
    ],
    espree: [
      "dynamic-import.js",
      "empty.js",
      "multi-types.js",
      "static-import.js",
      "re-export.js",
      "without-from.js",
      "non-type.js",
    ],
    meriyah: [
      "dynamic-import.js",
      "empty.js",
      "multi-types.js",
      "static-import.js",
      "re-export.js",
      "without-from.js",
      "non-type.js",
    ],
    typescript: [
      "empty.js",
      "multi-types.js",
      "static-import.js",
      "re-export.js",
      "without-from.js",
      "non-type.js",
    ],
  },
});
