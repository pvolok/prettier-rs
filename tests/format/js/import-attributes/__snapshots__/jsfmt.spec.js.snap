// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic-import.js [acorn] format 1`] = `
"Unexpected token (1:20)
> 1 | import("./foo.json", { with: { type: "json" } });
    |                    ^
  2 |"
`;

exports[`dynamic-import.js [espree] format 1`] = `
"Unexpected token , (1:20)
> 1 | import("./foo.json", { with: { type: "json" } });
    |                    ^
  2 |"
`;

exports[`dynamic-import.js [meriyah] format 1`] = `
"Expected ')' (1:20)
> 1 | import("./foo.json", { with: { type: "json" } });
    |                    ^
  2 |"
`;

exports[`dynamic-import.js format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import("./foo.json", { with: { type: "json" } });

=====================================output=====================================
import("./foo.json", { with: { type: "json" } });

================================================================================
`;

exports[`empty.js [acorn] format 1`] = `
"Unexpected token (2:33)
  1 | export * as foo from "foo.json"
> 2 | export * as bar from "bar.json" with { }
    |                                 ^
  3 | export * as baz from "baz.json" with { /* comment */ }
  4 |
  5 | import * as foo from "foo.json""
`;

exports[`empty.js [espree] format 1`] = `
"Unexpected token with (2:33)
  1 | export * as foo from "foo.json"
> 2 | export * as bar from "bar.json" with { }
    |                                 ^
  3 | export * as baz from "baz.json" with { /* comment */ }
  4 |
  5 | import * as foo from "foo.json""
`;

exports[`empty.js [meriyah] format 1`] = `
"Unexpected token: 'with' (2:36)
  1 | export * as foo from "foo.json"
> 2 | export * as bar from "bar.json" with { }
    |                                    ^
  3 | export * as baz from "baz.json" with { /* comment */ }
  4 |
  5 | import * as foo from "foo.json""
`;

exports[`empty.js [typescript] format 1`] = `
"';' expected. (2:33)
  1 | export * as foo from "foo.json"
> 2 | export * as bar from "bar.json" with { }
    |                                 ^
  3 | export * as baz from "baz.json" with { /* comment */ }
  4 |
  5 | import * as foo from "foo.json""
`;

exports[`empty.js format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
export * as foo from "foo.json"
export * as bar from "bar.json" with { }
export * as baz from "baz.json" with { /* comment */ }

import * as foo from "foo.json"
import * as bar from "bar.json" with { }
import * as baz from "baz.json" with { /* comment */ }

=====================================output=====================================
export * as foo from "foo.json";
export * as bar from "bar.json";
export * as baz from "baz.json" /* comment */;

import * as foo from "foo.json";
import * as bar from "bar.json";
import * as baz from "baz.json" /* comment */;

================================================================================
`;

exports[`multi-types.js [acorn] format 1`] = `
"Unexpected token (1:31)
> 1 | import json from "./foo.json" with { type: "json", type: "bar" };
    |                               ^
  2 |"
`;

exports[`multi-types.js [espree] format 1`] = `
"Unexpected token with (1:31)
> 1 | import json from "./foo.json" with { type: "json", type: "bar" };
    |                               ^
  2 |"
`;

exports[`multi-types.js [meriyah] format 1`] = `
"Unexpected token: 'with' (1:34)
> 1 | import json from "./foo.json" with { type: "json", type: "bar" };
    |                                  ^
  2 |"
`;

exports[`multi-types.js [typescript] format 1`] = `
"';' expected. (1:31)
> 1 | import json from "./foo.json" with { type: "json", type: "bar" };
    |                               ^
  2 |"
`;

exports[`multi-types.js format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import json from "./foo.json" with { type: "json", type: "bar" };

=====================================output=====================================
import json from "./foo.json" with { type: "json", type: "bar" };

================================================================================
`;

exports[`non-type.js [acorn] format 1`] = `
"Unexpected token (1:28)
> 1 | import foo from "foo.json" with { lazy: "true" };
    |                            ^
  2 |"
`;

exports[`non-type.js [espree] format 1`] = `
"Unexpected token with (1:28)
> 1 | import foo from "foo.json" with { lazy: "true" };
    |                            ^
  2 |"
`;

exports[`non-type.js [meriyah] format 1`] = `
"Unexpected token: 'with' (1:31)
> 1 | import foo from "foo.json" with { lazy: "true" };
    |                               ^
  2 |"
`;

exports[`non-type.js [typescript] format 1`] = `
"';' expected. (1:28)
> 1 | import foo from "foo.json" with { lazy: "true" };
    |                            ^
  2 |"
`;

exports[`non-type.js format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import foo from "foo.json" with { lazy: "true" };

=====================================output=====================================
import foo from "foo.json" with { lazy: "true" };

================================================================================
`;

exports[`re-export.js [acorn] format 1`] = `
"Unexpected token (1:44)
> 1 | export { default as foo2 } from "foo.json" with { type: "json" };
    |                                            ^
  2 | export * from "foo.json" with { type: "json" };
  3 | export * as foo3 from "foo.json" with { type: "json" };
  4 |"
`;

exports[`re-export.js [espree] format 1`] = `
"Unexpected token with (1:44)
> 1 | export { default as foo2 } from "foo.json" with { type: "json" };
    |                                            ^
  2 | export * from "foo.json" with { type: "json" };
  3 | export * as foo3 from "foo.json" with { type: "json" };
  4 |"
`;

exports[`re-export.js [meriyah] format 1`] = `
"Unexpected token: 'with' (1:47)
> 1 | export { default as foo2 } from "foo.json" with { type: "json" };
    |                                               ^
  2 | export * from "foo.json" with { type: "json" };
  3 | export * as foo3 from "foo.json" with { type: "json" };
  4 |"
`;

exports[`re-export.js [typescript] format 1`] = `
"';' expected. (1:44)
> 1 | export { default as foo2 } from "foo.json" with { type: "json" };
    |                                            ^
  2 | export * from "foo.json" with { type: "json" };
  3 | export * as foo3 from "foo.json" with { type: "json" };
  4 |"
`;

exports[`re-export.js format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
export { default as foo2 } from "foo.json" with { type: "json" };
export * from "foo.json" with { type: "json" };
export * as foo3 from "foo.json" with { type: "json" };

=====================================output=====================================
export { default as foo2 } from "foo.json" with { type: "json" };
export * from "foo.json" with { type: "json" };
export * as foo3 from "foo.json" with { type: "json" };

================================================================================
`;

exports[`static-import.js [acorn] format 1`] = `
"Unexpected token (1:31)
> 1 | import json from "./foo.json" with { type: "json" };
    |                               ^
  2 |"
`;

exports[`static-import.js [espree] format 1`] = `
"Unexpected token with (1:31)
> 1 | import json from "./foo.json" with { type: "json" };
    |                               ^
  2 |"
`;

exports[`static-import.js [meriyah] format 1`] = `
"Unexpected token: 'with' (1:34)
> 1 | import json from "./foo.json" with { type: "json" };
    |                                  ^
  2 |"
`;

exports[`static-import.js [typescript] format 1`] = `
"';' expected. (1:31)
> 1 | import json from "./foo.json" with { type: "json" };
    |                               ^
  2 |"
`;

exports[`static-import.js format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import json from "./foo.json" with { type: "json" };

=====================================output=====================================
import json from "./foo.json" with { type: "json" };

================================================================================
`;

exports[`without-from.js [acorn] format 1`] = `
"Unexpected token (1:14)
> 1 | import "foo" with { type: "json" }
    |              ^
  2 |"
`;

exports[`without-from.js [espree] format 1`] = `
"Unexpected token with (1:14)
> 1 | import "foo" with { type: "json" }
    |              ^
  2 |"
`;

exports[`without-from.js [meriyah] format 1`] = `
"Unexpected token: 'with' (1:17)
> 1 | import "foo" with { type: "json" }
    |                 ^
  2 |"
`;

exports[`without-from.js [typescript] format 1`] = `
"';' expected. (1:14)
> 1 | import "foo" with { type: "json" }
    |              ^
  2 |"
`;

exports[`without-from.js format 1`] = `
====================================options=====================================
parsers: ["babel", "typescript"]
printWidth: 80
                                                                                | printWidth
=====================================input======================================
import "foo" with { type: "json" }

=====================================output=====================================
import "foo" with { type: "json" };

================================================================================
`;
